dogpile/cache/region.py

```diff
@@ -327,6 +328,10 @@ def generate_keys(*args):
      by the backend.  Defaults to ``None``, in which case the
      deserializer recommended by the backend will be used.   Typical
      deserializers include ``pickle.dumps`` and ``json.dumps``.
+     Deserializers can raise a :class:`.api.CantDeserializeException` if they
+     are unable to deserialize the value from the backend, indicating
+     deserialization failed and that caching should proceed to re-generate
+     a value.
```

Comment:
**mike bayer** (zzzeek) wrote:

then maybe another line with rationale: "this can allow an application that's been updated to gracefully re-cache old items that were persisted from a previous version of the application"

since I had to figure out why we want this

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/dogpile.cache/+/4569